pkg/virtctl/spice/spice.go

```diff
@@ -0,0 +1,102 @@
+package spice
+
+import (
+	"errors"
+	"fmt"
+	flag "github.com/spf13/pflag"
+	"io/ioutil"
+	"k8s.io/client-go/tools/clientcmd"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+const FLAG = "spice"
+const SPICE_FILENAME = "console.vv"
+const VM_SPICE_URL = "http://%s/apis/kubevirt.io/v1alpha1/namespaces/default/vms/%s/spice"
+
+type Spice struct {
+}
+
+func DownloadSpice(vm, host string) (string, error) {
+	var path string = fmt.Sprintf(VM_SPICE_URL, host, vm)
+	req, err := http.NewRequest("GET", path, nil)
```

Comment:
We can use the rest client from kubernetes here:


```golang
body, err := restClient.Get().
  Resource("vms").SetHeader("Accept", rest.MIME_INI).
  SubResource("spice").
  Namespace(kubev1.NamespaceDefault).
  Name(vm.GetObjectMeta().GetName()).Do().Raw()
```

You can get a rest client which understands the VM TPR like this:

```golang
restClient, err := kubecli.GetRESTClientFromFlags(master, kubeconfig)
```

This also allows us to reuse the authentication information from kubeconfig.